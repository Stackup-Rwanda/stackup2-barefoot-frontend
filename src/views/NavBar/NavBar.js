import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Link from '@material-ui/core/Link';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  navlink: {
    color: theme.palette.primary.main,
  },
}));

const NavBar = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static" className="navbar">
        <Toolbar className="navbar-content">
          <div className="brand">
            <h2 className="brand-barefoot">Barefoot</h2>
            <h2 className="brand-nomad">Nomad</h2>
          </div>
          <div className="brand">
            <Link component={RouterLink} to="/login" color="primary" className="navlink">Login</Link>
            <Link href="/signup" color="textPrimary" className="navlink navlink-signup">Create account</Link>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default NavBar;
