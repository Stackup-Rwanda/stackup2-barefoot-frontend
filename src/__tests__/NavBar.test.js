/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import NavBar from '../views/NavBar/NavBar';

describe('<NavBar/>', () => {
  afterEach(cleanup);
  it('Should match the NavBar component snapshot', () => {
    const tree = renderer.create(<Router><NavBar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
