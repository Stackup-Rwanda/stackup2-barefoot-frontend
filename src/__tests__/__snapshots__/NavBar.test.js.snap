// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> Should match the NavBar component snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary navbar MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular navbar-content MuiToolbar-gutters"
    >
      <div
        className="brand"
      >
        <h2
          className="brand-barefoot"
        >
          Barefoot
        </h2>
        <h2
          className="brand-nomad"
        >
          Nomad
        </h2>
      </div>
      <div
        className="brand"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover navlink MuiTypography-colorPrimary"
          href="/login"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Login
        </a>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover navlink navlink-signup MuiTypography-colorTextPrimary"
          href="/signup"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Create account
        </a>
      </div>
    </div>
  </header>
</div>
`;
